alphastats/DataSet_Preprocess.py

```diff
@@ -30,45 +31,58 @@ def preprocess_print_info(self):
         print(pd.DataFrame(self.preprocessing_info.items()))
 
     def _remove_na_values(self, cut_off):
+        if (
+            self.preprocessing_info.get("Missing values were removed")
+            and self.preprocessing_info.get("Data completeness cut-off") == cut_off
+        ):
+            logging.info("Missing values have already been filtered.")
+            st.warning(
+                "Missing values have already been filtered. To apply another cutoff, reset preprocessing."
+            )
+            return
         cut = 1 - cut_off
-        limit = self.mat.shape[0] * cut
-        
+
+        num_samples, num_proteins = self.mat.shape
+        limit = num_samples * cut
+
+        self.mat.replace(0, np.nan, inplace=True)
```

Comment:
in principle, there's a danger here that we consider something that is actually "zero" as "missing" (should not be a problem in real life as these are floating point numbers I guess) .. still, do we have control over that upstream component that sets the zeroes initially? If so, one could try to change the zeroes to NaN there ..